### DISCUSSION TITLE
how to improve the parallelism of queries

### DISCUSSION THREAD

**Discussion by qicaiyulin** [databendlabs/databend#7607] [Q&A] [answered]
When a cluster has multiple query nodes, whether the sql query task of a query node can be executed concurrently on multiple query nodes?

**Comment by bohutang** [comment, marked as answer]
Yes.
Currently, `SELECT` and `INSERT` in Databend are distributed among cluster nodes. Others we are still improving, such as the `COPY` command.

The main code is in:
https://github.com/datafuselabs/databend/blob/d6b86315015f052ceba3741b57391e6dbf87e7fb/src/query/service/src/optimizers/optimizer_scatters.rs
